plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -311,9 +313,20 @@ We keep the counters for each step in the first indices, so we can index them si
 the 'Enum' instance of 'StepKind', and the total counter in the last index.
 -}
 
+-- So that we don't need to update 'NumberOfStepCounters' manually, which would be extremely
+-- error-prone and has caused a bug in the past.
+type CountConstructorsEnum :: (GHC.Type -> GHC.Type) -> Nat
+type family CountConstructorsEnum rep where
```

Comment:
I was going to suggest that we should put this somewhere else since it's not very Cek-specific, but I don't know where would be a sensible place for it.  Maybe `StepCounter.hs`?

Reply:
I'd like to keep it in there in case the structure of `StepKind` somehow changes. Like, what if it becomes parameterized for whatever reason? Currently it's written specifically with the structure of the existing `StepKind` in mind.